Rules/CommonScripts/RedBarrier.as

```diff
@@ -1,291 +1,245 @@
-			cbs.write_f32(y2);
-
-			this.SendCommand(this.getCommandID("set_barrier_pos"), cbs, true);
-		}
+		IS_BARRIER_SET = true;
 		return;
 	}
 
-	if (shouldBarrier(this))
-	{
-		if (!barrier_set)
-		{
-			if (barrier_timer < barrier_wait)
-			{
-				barrier_timer++;
-			}
-			else
-			{
-				barrier_set = true;
-				addBarrier();
-			}
-		}
+	LoadConfigVars();
 
-		f32 x1, x2, y1, y2;
-		getBarrierPositions(x1, x2, y1, y2);
-		const f32 middle = x1 + (x2 - x1) * 0.5f;
+	SetBarrierPosition(this);
 
-		CBlob@[] blobsInBox;
-		if (getMap().getBlobsInBox(Vec2f(x1, y1), Vec2f(x2, y2), @blobsInBox))
-		{
-			for (uint i = 0; i < blobsInBox.length; i++)
-			{
-				CBlob @b = blobsInBox[i];
-				if (b.getTeamNum() < 100 || b.hasTag("no barrier pass"))
-				{
-					Vec2f pos = b.getPosition();
-
-					//players clamped to edge
-					if (b.getPlayer() !is null)
-					{
-						Vec2f pos = b.getPosition();
-						if (pos.x >= x1 && pos.x <= x2)
-						{
-							Vec2f vel = b.getVelocity();
-							float margin = 0.01f;
-							float vel_base = 0.01f;
-							if (pos.x < middle)
-							{
-								pos.x = Maths::Min(x1 - margin, pos.x) - margin;
-								vel.x = Maths::Min(-vel_base, -Maths::Abs(vel.x));
-							}
-							else
-							{
-								pos.x = Maths::Max(x2 + margin, pos.x) + margin;
-								vel.x = Maths::Max(vel_base, Maths::Abs(vel.x));
-							}
-							b.setPosition(pos);
-							b.setVelocity(vel);
-						}
-					}
-					//other objects pushed softly (annoying for players apparently)
-					else
-					{
-						f32 f = b.getMass() * 2.0f;
-
-						if (pos.x < middle)
-						{
-							b.AddForce(Vec2f(-f, -f * 0.1f));
-						}
-						else
-						{
-							b.AddForce(Vec2f(f, -f * 0.1f));
-						}
-					}
-				}
-			}
-		}
-	}
-	else
+	const int playerCount = getPlayerCount();
+	for (int a = 0; a < playerCount; a++)
 	{
-		if (barrier_set)
-		{
-			removeBarrier();
-			barrier_set = false;
-			barrier_timer = 0;
-		}
+		CPlayer@ player = getPlayer(a);
+
+		if (player is null)
+			continue;
+
+		SyncToPlayer(this, player);
+
+		if (SYNC_CUSTOM_VARS)
+			SyncVarsToPlayer(this, player);
 	}
 }
 
-void onInit(CRules@ this)
+void onNewPlayerJoin(CRules@ this, CPlayer@ player)
 {
-	this.addCommandID("set_barrier_pos");
-	onRestart(this);
-}
+	if (!isServer() && !shouldBarrier(this))
+		return;
 
-void onRestart(CRules@ this)
-{
-	barrier_set = false;
-	done_sync = false;
-	barrier_timer = 0;
+	SyncToPlayer(this, player);
+
+	if (SYNC_CUSTOM_VARS)
+		SyncVarsToPlayer(this, player);
 }
 
-void onRender(CRules@ this)
+void onTick(CRules@ this)
 {
-	if (!done_sync) return;
+	if (!shouldBarrier(this))
+	{
+		IS_BARRIER_SET = false;
+		return;
+	}
+
+	CMap@ map = getMap();
+
+	const u16 x1 = this.get_u16("barrier_x1");
+	const u16 x2 = this.get_u16("barrier_x2");
+	const u16 middle = (x1 + x2) * 0.5f;
 
-	if (shouldBarrier(this))
+	CBlob@[] blobsInBox;
+	if (map.getBlobsInBox(Vec2f(x1, 0), Vec2f(x2, map.tilemapheight * map.tilesize), @blobsInBox))
```

Comment:
This makes it so that you can shoot things over the red barrier. It might be a problem for SmallCTF.

Reply:
Nice catch. 50 or so tiles extension is much more than enough to catch all objects.
```suggestion
	if (map.getBlobsInBox(Vec2f(x1, -50 * map.tilesize), Vec2f(x2, map.tilemapheight * map.tilesize), @blobsInBox))
```